test/models/collection_test.rb

```diff
@@ -0,0 +1,21 @@
+require 'test_helper'
+
+class CollectionTest < ActiveSupport::TestCase
+  test 'should create a new collection' do
+    collection = Collection.new(name: 'Test', user_id: 1, city_id: 1)
+    assert collection.valid?
+    assert collection.save
+  end
+  test 'invlaid without collection name' do
+    collection = Collection.new(user_id: 1, city_id: 1)
+    collection.name = nil
+    refute collection.valid?, 'Collection is invalid without name'
+    assert_includes collection.errors[:name], "Collection name can't be blank"
+  end
+
+  test "collection can't be created without registered user" do
```

Comment:
There is also an opportunity to make a test for creating a collection without a city, assuming the `presence: true` validation on the city is to stay.